plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2003,23 +2005,28 @@ private void createEmptyFiles(ConnectorSession session, Location path, Table tab
         }
 
         for (String fileName : fileNames) {
-            Location location = path.appendPath(fileName);
-            fileWriterFactories.stream()
-                    .map(factory -> factory.createFileWriter(
-                            location,
-                            ImmutableList.of(),
-                            format,
-                            HiveCompressionCodec.NONE,
-                            schema,
-                            session,
-                            OptionalInt.empty(),
-                            NO_ACID_TRANSACTION,
-                            false,
-                            WriterKind.INSERT))
-                    .flatMap(Optional::stream)
-                    .findFirst()
-                    .orElseThrow(() -> new TrinoException(HIVE_UNSUPPORTED_FORMAT, "Writing not supported for " + format))
-                    .commit();
+            try {
+                Location location = path.appendPath(fileName);
+                fileWriterFactories.stream()
+                        .map(factory -> factory.createFileWriter(
+                                location,
+                                ImmutableList.of(),
+                                format,
+                                HiveCompressionCodec.NONE,
+                                schema,
+                                session,
+                                OptionalInt.empty(),
+                                NO_ACID_TRANSACTION,
+                                false,
+                                WriterKind.INSERT))
+                        .flatMap(Optional::stream)
+                        .findFirst()
+                        .orElseThrow(() -> new TrinoException(HIVE_UNSUPPORTED_FORMAT, "Writing not supported for " + format))
+                        .commit();
+            }
+            catch (InvalidHiveSchemaException e) {
```

Comment:
This seems like the wrong location.  I'd put this in LinePageSourceFactory.  If we make this exception checked it will be clear where the catch needs to go.